main.c

```diff
@@ -79,11 +80,26 @@ static void finish(struct mako_state *state) {
 	wl_list_for_each_safe(notif, tmp, &state->history, link) {
 		destroy_notification(notif);
 	}
+
+	struct mako_surface *surface, *stmp;
+	wl_list_for_each_safe(surface, stmp, &state->surfaces, link) {
+		destroy_surface(surface);
+	}
 	finish_event_loop(&state->event_loop);
 	finish_wayland(state);
 	finish_dbus(state);
 }
 
+static void init_surfaces(struct mako_state *state)
+{
+	wl_list_init(&state->surfaces);
+	struct mako_surface_config *config;
+	wl_list_for_each(config, &state->config.surfaces, link) {
+		/* We don't need the return value right now */
```

Comment:
Same question as above about how we should handle allocation failures here.

Reply:
In practice I'd probably abort the daemon if it's the default `"(root)"` surface, and otherwise continue on, since it'll still work, just not quite as expected.

Reply:
See https://github.com/emersion/mako/pull/264